network/server_test.go

```diff
@@ -793,6 +793,10 @@ func TestSubscribe(t *testing.T) {
 		// cancel before emitting
 		cancel()
 
+		select {
+		case <-ctx.Done():
+		}
```

Comment:
IMO, it is not necessary having select with a single case.

```suggestion
    <-ctx.Done()
```